[skia,dawn] Fix support for BSDs

cqundefine opened the pull request:
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] SHA512s are updated for each updated download.
- [x] The "supports" clause reflects platforms that may be fixed by this new version.
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

cqundefine committed:
[skia] Specify the license

cqundefine committed:
[skia] Add a `dng` feature

cqundefine committed:
[skia] Misc BSD fixes

cqundefine committed:
[dawn] Fix support for BSDs

xiaozhuai commented:
I haven't tested dawn on the BSD platform before, and I'm glad someone can complete this.
Would you please check if `vcpkg-ci-dawn` works on bsd?

cqundefine committed:
[skia] Use pkgconfig to find GL

This fixes an issue when OpenGL libs and headers are in non-standard
locations like /usr/local/...

cqundefine committed:
[vcpkg-ci-dawn] Fix the test on BSD systems

cqundefine commented:
> Would you please check if `vcpkg-ci-dawn` works on bsd?

It does, I did have to change one ifdef, but that's the only change necessary.

JavierMatosD commented on ports/skia/use-pkgconfig-to-find-gl.patch:
```diff
@@ -0,0 +1,15 @@
+       ]
+-      libs += [ "GL" ]
++      deps += [
++        "//third_party/gl"
```
vendored dependency?
